src/bridges/yearn/YearnBridge.sol

```diff
@@ -0,0 +1,205 @@
+     * @param _rollupProcessor Address of rollup processor
+     */
+    constructor(address _rollupProcessor) BridgeBase(_rollupProcessor) {}
+
+    receive() external payable {}
+
+    /**
+     * @notice Set all the necessary approvals for all the latests vaults for the tokens supported by Yearn
+     */
+    function preApproveAll() external {
+        uint256 numTokens = YEARN_REGISTRY.numTokens();
+        for (uint256 i; i < numTokens; ) {
+            address token = YEARN_REGISTRY.tokens(i);
+            address vault = YEARN_REGISTRY.latestVault(token);
+            _preApprove(vault, token);
+
+            unchecked {
+                ++i;
+            }
+        }
+    }
+
+    /**
+     * @notice Set all the necessary approvals for a given vault and its underlying.
+     * @dev The vault must be a valid Yearn vault, registered in the registry
+     * @param _vault - vault address
+     */
+    function preApprove(address _vault) external {
+        address token = IYearnVault(_vault).token();
+        uint256 numVaultsForToken = YEARN_REGISTRY.numVaults(token);
+        for (uint256 i; i < numVaultsForToken; ) {
+            address vault = YEARN_REGISTRY.vaults(token, i);
+            if (vault == _vault) {
+                _preApprove(vault, token);
+                return;
+            }
+            unchecked {
+                ++i;
+            }
+        }
+        revert InvalidVault();
+    }
+
+    /**
+     * @notice A function which will allow the user to deposit/withdraw a given amount of a given token to a given vault. When depositing to a Yearn vault,
+     * the user deposits the underlying (or want) token as input (ex: DAI). When withdrawing from a Yearn vault, the user asks to withdraws some shares of the
+     * Yearn Vaults, where input is the yvToken (ex: yvDAI) and receive the underlying token as output (ex: DAI).
+     * @param _inputAssetA - ERC20 token to deposit or Yearn Vault ERC20 token to withdraw
+     * @param _outputAssetA - Yearn Vault ERC20 token to receive on deposit or ERC20 token to receive on withdraw
+     * @param _inputValue - Amount to deposit or withdraw
+     * @param _interactionNonce - Unique identifier for this DeFi interaction
+     * @param _auxData - Special value to indicate a deposit (0) or a withdraw (1)
+     * @return outputValueA - the amount of output asset to return
+     */
+    function convert(
+        AztecTypes.AztecAsset memory _inputAssetA,
+        AztecTypes.AztecAsset memory,
+        AztecTypes.AztecAsset memory _outputAssetA,
+        AztecTypes.AztecAsset memory,
+        uint256 _inputValue,
+        uint256 _interactionNonce,
+        uint64 _auxData,
+        address
+    )
+        external
+        payable
+        override(BridgeBase)
+        onlyRollup
+        returns (
+            uint256 outputValueA,
+            uint256,
+            bool
+        )
+    {
+        if (_inputValue == 0) {
+            revert ErrorLib.InvalidInputAmount();
```

Comment:
This check is unnecessary because we check this in a `RollupProcessor` as well so please remove it. Normally this would be a good practice to have it but since we control the source this is not necessary.
<img width="687" alt="image" src="https://user-images.githubusercontent.com/13470840/182095215-d1a04183-adfb-4523-b198-83264406c465.png">